src/commands.rs

```diff
@@ -1102,6 +1106,27 @@ struct GlobalArgs {
     at_operation: String,
 }
 
+impl GlobalArgs {
+    fn merge_from(&mut self, other: &GlobalArgs) {
+        // Destructure `other`, so we're forced to update this code if new fields are
+        // added in the future.
+        let GlobalArgs {
+            repository,
+            no_commit_working_copy,
+            at_operation,
+        } = other;
+        if self.repository.is_some() {
```

Comment:
`s/self\.//` ;) 
I meant (other.)`repository.is_some()` so the `other` wins.

If `self` should win, the other `if (no self.)<field>` condition can be flipped.

Reply:
> `s/self\.//` ;) I meant (other.)`repository.is_some()` so the `other` wins.

Haha, I'm sorry.

> If `self` should win, the other `if (no self.)<field>` condition can be flipped.

Yes, that's what I had intended. I've updated it to work that way. I've also updated and added tests to check it better, given that I've messed it up twice already 😄